client/state/comments/reducer.js

```diff
@@ -113,14 +114,20 @@ export function items( state = {}, action ) {
  * @returns {Object} new redux state
  */
 export function totalCommentsCount( state = {}, action ) {
+	const { siteId, postId } = action;
+	const stateKey = getStateKey( siteId, postId );
+
 	switch ( action.type ) {
 		case COMMENTS_COUNT_RECEIVE:
-			const { siteId, postId } = action;
-			const stateKey = getStateKey( siteId, postId );
 			return {
 				...state,
 				[ stateKey ]: action.totalCommentsCount
 			};
+		case COMMENTS_COUNT_INCREMENT:
+			return {
+				...state,
+				[ stateKey ]: state[ stateKey ] + 1
+			};
```

Comment:
yes it will. But we'll have to add a decrement action anyway if the update fails.